src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Table.cs

```diff
@@ -59,27 +59,44 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
             var rowType = DType.EmptyRecord;
             for (var i = 0; i < argTypes.Length; i++)
             {
+                var isUnionError = false;
                 var argType = argTypes[i];
                 if (!argType.IsRecord)
                 {
                     errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrNeedRecord);
                     isValid = false;
                 }
-                else if (!rowType.CanUnionWith(argType))
-                {
-                    errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrIncompatibleRecord);
-                    isValid = false;
-                }
                 else
```

Comment:
>else [](http://example.com/codeflow?start=16&length=4)

This change should be done under a new feature flag

Reply:
So, we are not breaking anything by restricting the behavior. We are allowing `Table()` now to operate in a more "loosen" way. The changes will give an different error message though. With that been said, do you still think this needs a new flag?

Reply:
In this case we shouldn't need a flag